### ISSUE TITLE
Sparse nonnegative Tucker decomposition

### ISSUE THREAD

**Issue by montyvesselinov** [tensorly/tensorly#79]
Do you plan to add Sparse nonnegative Tucker decomposition?

Julia version of the code is here:

https://github.com/yunjhongwu/TensorDecompositions.jl/blob/master/src/spnntucker.jl

Here is also the original paper.

Xu, Y. (2015). Alternating proximal gradient method for sparse nonnegative Tucker decomposition. Mathematical Programming Computation, 7(1), 39-70.

**Comment by JeanKossaifi** [comment]
Yes, we want to have sparse decomposition - we are in the process of adding sparse tensor support (#76, #77). 

We welcome additions of new decompositions if you want to take a crack at it!

**Comment by cohenjer** [comment]
@JeanKossaifi I think @montyvesselinov was thinking about a nonnegative Tucker decomposition with a sparsifying regularization, rather than the NTD of a sparse tensor. At least the mentioned research article deals with the former.

We (a PhD student and me) have a local implementation of (sparse) NTD which we will soon format for a pull-request in TensorLy. It uses sparse nonnegative least squares as a subroutine rather than proximal gradient, but the two methods are closely related. Stay tuned!
